natlas-server/Dockerfile

```diff
@@ -0,0 +1,34 @@
+FROM python:3.6.0 as build
+
+WORKDIR /opt/natlas/natlas-server
+COPY requirements.txt /opt/natlas/natlas-server
+
+RUN pip3 --disable-pip-version-check --no-cache-dir install --root=/install -r requirements.txt \
+	&& rm -rf requirements.txt
+
+COPY . /opt/natlas/natlas-server
+
+RUN python3 -m compileall .
+
+# Build final image
+FROM python:3.6.0-slim
+
+RUN export DEBIAN_FRONTEND=noninteractive \
+	&& apt-get update \
+	&& apt-get install --no-install-recommends -qy libsqlite3-0
```

Comment:
You'll want to structure your RUN layers in a way that minimizes them. Having lots of layers is not so good.

Reply:
Having lots of layers means that each layer can be cached if it doesn't change, which also means that building new docker images when source updates but dependencies don't update will only rebuild the layers that need to be rebuilt, which will be very small layers.

Reply:
I typically do no-cache when doing dev, and if you want to make dev builds more "responsive" you can always mount the code directory into the container.  Its nbd, but something to keep an eye on if we start seeing issues.

Reply:
This is a multi stage Dockerfile so collapsing the first stage into a single RUN command won't help us and instead make it build time worse because we have to do more cache invitations.

I do agree with mounting the dev directory into the container. I'd like to see a docker-compose file for developers that mounts the directory.

The second stage is the image that would get deployed to the registry and is compacted.

Reply:
Yeah, Multistage builds are good stuff. I used them in a couple builds. Build caches are great for local stuff. I tend to not worry about it, because of the fact that most solutions like travis-ci/circle-ci gitlab don't utilize them fully.

Reply:
I'm going to merge the PR into `main` and then if someone wants to add the docker-compose in another PR then we can handle that as well.